src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,509 @@
+        terrain_normal = jax.vmap(self.terrain.normal)(position[:, 0], position[:, 1])
+        n_collidable_points = model.kin_dyn_parameters.contact_parameters.point.shape[0]
+
+        # Compute the activation state of the collidable points
+        inactive_collidable_points, (delta, delta_dot) = RigidContacts.detect_contacts(
+            W_o_C=position,
+            W_o_dot_C=velocity,
+            terrain_height=terrain_height,
+            terrain_normal=terrain_normal,
+        )
+
+        delassus_matrix = RigidContacts._delassus_matrix(M=M, J_WC=J_WC)
+        # Make it symmetric if not
+        delassus_matrix = jax.lax.cond(
+            jnp.allclose(delassus_matrix, delassus_matrix.T),
+            lambda G: G,
+            lambda G: (G + G.T) / 2,
+            delassus_matrix,
+        )
+        # Add regularization for better numerical conditioning
+        delassus_matrix = delassus_matrix + 1e-6 * jnp.eye(delassus_matrix.shape[0])
+
+        references = js.references.JaxSimModelReferences.build(
+            model=model,
+            data=data,
+            velocity_representation=data.velocity_representation,
+            link_forces=link_external_forces,
+        )
+
+        with references.switch_velocity_representation(VelRepr.Mixed):
+            nu_dot_free_mixed = RigidContacts._compute_mixed_nu_dot_free(
+                model, data, references=references
+            )
+
+        free_contact_acc = RigidContacts._linear_acceleration_of_collidable_points(
+            model,
+            data,
+            nu_dot_free_mixed,
+        ).flatten()
+        # Compute stabilization term
+        baumgarte_term = RigidContacts._compute_baumgarte_stabilization_term(
+            inactive_collidable_points=inactive_collidable_points,
+            delta=delta,
+            delta_dot=delta_dot,
+            K=self.parameters.K,
+            D=self.parameters.D,
+        ).flatten()
+        free_contact_acc -= baumgarte_term
+
+        # Setup optimization problem
+        Q = delassus_matrix
+        q = free_contact_acc
+        G = RigidContacts._compute_ineq_constraint_matrix(
+            inactive_collidable_points=inactive_collidable_points, mu=self.parameters.mu
+        )
+        h = RigidContacts._compute_ineq_bounds(n_collidable_points=n_collidable_points)
+        A = jnp.zeros((0, 3 * n_collidable_points))
+        b = jnp.zeros((0,))
+
+        # Solve the optimization problem
+        solution, *_ = qpax.solve_qp(Q=Q, q=q, A=A, b=b, G=G, h=h)
+
+        f_C_lin = solution.reshape(-1, 3)
+
+        # Transform linear contact forces to 6D
+        CW_f_C = jnp.hstack(
+            (
+                f_C_lin,
+                jnp.zeros((f_C_lin.shape[0], 3)),
+            )
+        )
+
+        # Transform the contact forces to inertial-fixed representation
+        W_f_C = RigidContacts._compute_link_forces_inertial_fixed(
+            CW_f_C=CW_f_C, W_H_C=W_H_C
+        )
+
+        return (W_f_C, ())
```

Comment:
```suggestion
        return W_f_C, ()
```

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e